### DISCUSSION TITLE
ConsumerStatefulWidget vs Consumer

### DISCUSSION THREAD

**Discussion by cyb9701** [rrousselGit/riverpod#2543] [Q&A] [answered]
There are several widgets in one class.
If ref is used in only one widget among many widgets,
Is it more efficient to create that widget as a Consumer widget?
I wonder if it would be more efficient to extend the entire class to ConsumerStatefulWidget.
Which method is better for app performance? I'd be grateful if you let me know if there's a better code.

**Comment by rrousselGit** [comment, marked as answer]
There's no real performance difference between these two classes. It's about whether you want a StatefulWidget or not.
